Add facility for automatically typing passwords via XMonad.Prompt.Pass, and handle special characters in pass labels

NickHu opened the pull request:
### Description

Adds a new function to spawn a pass prompt which will use `xdotool` to
type in a password, bypassing the clipboard. Also incorporate some shell
escapes to properly handle pass labels with spaces and special
characters in them.

### Checklist

  - [x] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - [x] I tested my changes with [xmonad-testing](https://github.com/xmonad/xmonad-testing)

  - [x] I updated the `CHANGES.md` file

geekosaur commented:
First, you broke backward compatibility by changing all the Monad to Applicative that you could. This is why Travis complained. Please stick to backward compatible interfaces.

Second: you do realize that, while the `xdotool` command is running, the password will be visible in a `ps` listing?

NickHu commented:
I hope my last commit addresses your concerns - I found the `--file` parameter for `xdotool` here https://unix.stackexchange.com/questions/237608/does-xdotool-have-a-log

byorgey commented:
@geekosaur How does this look to you now?  It looks OK to me, but you know a lot more about security than I do.

NickHu commented:
I've rebased this onto master; I feel like this is ready to be merged.

byorgey commented:
Sorry for the long hiatus, and thanks for the patch!

NickHu commented:
No problem; any chance you could look at my other pr? https://github.com/xmonad/xmonad-contrib/pull/181
